ARCHITECTURE.md

```diff
@@ -158,10 +173,115 @@ This button with type submit inside a form will make it submit on ENTER key pres
+
+
+## Asset Handling
+
+
+Images uploaded by creators in Geyser are automatically resized into three distinct sizes: small, medium, and large. These images share the same URL, differentiated by postfixes such as `image_small.webp`, `image_medium.webp` and `image_large.webp`.
+
+For instance, if a user uploads an image named 'profile.png', it's uploaded, and the returned value is the small image in the format `https://geyserbucket...../profile_image_small.webp`. This process is automatic, allowing us to add the image directly to the database without manual manipulation.
+
+However, there are situations where a larger image size is required, like in a project thumbnail popup. In such cases, the small image URL needs to be converted to a large image URL, i.e., `https://geyserbucket...../profile_image_large.webp`.
+
+To facilitate this, we've provided functions in the utils folder: `toLargeImageUrl`, `toMediumImageUrl`, and `toSmallImageUrl`. These functions can be used to display images of different sizes as necessary. Even if the incoming image size is unknown, these functions act as filters, ensuring the correct image size renders.
+
+
+
+## Architecture Goal 
+
+The geyser-app is currently being restructured to improve maintainability, readability, and performance.
+Guided by Domain Driven Design (DDD) principles and the Horizontal Split principle, we'll reorganize all pages and components into Modules.
+
+### Project Structure
+The base structure of the app would be introduced with the following folders among the existing ones:
+- /src/shared
+- /src/modules
+
+#### Shared
+This would contain all of the business-logic-less ui components, contants, helpers and utils, that would be required by the modules.
+
+A general structure looks like:
+1. `shared` package
+    1. `components` 
+        1. `ui`
+        2. `layout`
+        3. `input`
+            1. `form`
+            2. `elements`
+    2. `constants`
+    3. `utils`
+
+#### Modules
+We will have a module per domain, and a domain would consist of multiple pages, along with other tools. Based on the nature of pages and the difference in complexity. We have `views` inside `pages` that replicate a `pages` structure, so we can go as deeply nested as needed.
+
+ A general structure for modules looks like:
+
+1.  `states` *( state manipulation / atoms )*
+2.  `hooks` *( logic, and communication backend )*
+3.   `components` *( ui and layout components )*
+4.  `graphQL` *( fragment, queries definition )*
+5.  `types` *( fragment types, query and mutation hooks )*
+6.  `pages` 
+	1.  `states`
+	2.  `hooks`
+	5.  `components`
+	6. ` views`
+        1.  *…recursive ( pages )*
+
+Below are the planned modules and the pages that would go in them based on the pages that are currently in the app. 
+
+1.  **project**
```

Comment:
I think this has more detail than needed. Listing the core modules would be enough, as you already highlighted the internal structure of the modules above. If you want you can add a short description about the modules as well.